File: go/neetcode/array_hashing/1_two_sum.go
package array_hashing

import "fmt"

func twoSum(nums []int, target int) []int {
	mapping_nums := make(map[int]int)

	for pos, num := range nums {
		mapping_nums[num] = pos
	}

	for i := 0; i < len(nums); i++ {
		want := target - nums[i]
		if mapping_nums[want] != 0 && i != mapping_nums[want] {
			return []int{i, mapping_nums[want]}
		}
	}
	return []int{}

}

func twoSum_2(nums []int, target int) []int {
	return []int{}
}

func Execute_twoSum() {
	nums := []int{2, 7, 11, 15}
	target := 9
	ans := twoSum(nums, target)
	fmt.Println(ans)

}

File: py/1_two_sum.py
from typing import List


def twoSum_brute(nums: List[int], target: int) -> List[int]:
    for i, val in enumerate(nums):
        print(i, val)


def twoSum(nums: List[int], target: int) -> List[int]:
    di = {}
    for i, val in enumerate(nums):
        want = target - val
        want_index = di.get(want)
        if want_index != None and i != want_index:
            return i, want_index
        di[val] = i


nums = [2, 7, 11, 15]
target = 6
ans = twoSum_brute(nums, target)
print(ans)
# ans = twoSum(nums, target)
# print(ans)
